receiver/aerospikereceiver/internal/model/aerospike.go

```diff
@@ -0,0 +1,261 @@
+	FabricConnectionsOpened    *int64 `json:"fabric_connections_opened"`
+	HeartbeatConnections       *int64 `json:"heartbeat_connections"`
+	HeartbeatConnectionsClosed *int64 `json:"heartbeat_connections_closed"`
+	HeartbeatConnectionsOpened *int64 `json:"heartbeat_connections_opened"`
+	SystemFreeMemPct           *int64 `json:"system_free_mem_pct"`
+}
+
+// NamespaceInfo is specific information about a single namespace on a node
+type NamespaceInfo struct {
+	Name                    string `json:"name"`
+	Node                    string `json:"node_name"`
+	DeviceAvailablePct      *int64 `json:"device_available_pct"`
+	DeviceUsedBytes         *int64 `json:"device_used_bytes"`
+	HighWaterDiskPct        *int64 `json:"high-water-disk-pct"`
+	HighWaterMemoryPct      *int64 `json:"high-water-memory-pct"`
+	MemoryFreePct           *int64 `json:"memory_free_pct"`
+	MemoryUsedDataBytes     *int64 `json:"memory_used_data_bytes"`
+	MemoryUsedIndexBytes    *int64 `json:"memory_used_index_bytes"`
+	MemoryUsedSetIndexBytes *int64 `json:"memory_used_set_index_bytes"`
+	MemoryUsedSIndexBytes   *int64 `json:"memory_used_sindex_bytes"`
+	ScanAggrAbort           *int64 `json:"scan_aggr_abort"`
+	ScanAggrComplete        *int64 `json:"scan_aggr_complete"`
+	ScanAggrError           *int64 `json:"scan_aggr_error"`
+	ScanBasicAbort          *int64 `json:"scan_basic_abort"`
+	ScanBasicComplete       *int64 `json:"scan_basic_complete"`
+	ScanBasicError          *int64 `json:"scan_basic_error"`
+	ScanOpsBgAbort          *int64 `json:"scan_ops_bg_abort"`
+	ScanOpsBgComplete       *int64 `json:"scan_ops_bg_complete"`
+	ScanOpsBgError          *int64 `json:"scan_ops_bg_error"`
+	ScanUdfBgAbort          *int64 `json:"scan_udf_bg_abort"`
+	ScanUdfBgComplete       *int64 `json:"scan_udf_bg_complete"`
+	ScanUdfBgError          *int64 `json:"scan_udf_bg_error"`
+}
+
+// InfoResponse is the map returned by the INFO command
+type InfoResponse map[string]string
+
+// ParseInfo creates NodeInfo using values present in the INFO response
+func ParseInfo(i InfoResponse) *NodeInfo {
+	nodeInfo := NodeInfo{}
+
+	if v, ok := i["node"]; ok {
+		nodeInfo.Name = v
+	}
+	if v, ok := i["services"]; ok {
+		nodeInfo.Services = splitFields(v)
+	}
+	if v, ok := i["namespaces"]; ok {
+		nodeInfo.Namespaces = splitFields(v)
+	}
+	// If statistics aren't present, keep Statistics a nil pointer
+	if _, ok := i[statisticsKey]; ok {
+		nodeInfo.Statistics = parseNodeStats(valueToMap(i, statisticsKey))
+	}
+
+	return &nodeInfo
+}
+
+// ParseNamespaceInfo creates NamespaceInfo using values present in the INFO response
+func ParseNamespaceInfo(i InfoResponse, namespace string) *NamespaceInfo {
+	infoMap := valueToMap(i, NamespaceKey(namespace))
+	namespaceInfo := NamespaceInfo{
+		Name: namespace,
+	}
+
+	// No point in checking an empty map
+	if len(infoMap) == 0 {
+		return &namespaceInfo
+	}
+
+	if v := parseInt(infoMap, "memory_used_data_bytes"); v != nil {
```

Comment:
Can this be refactored to a function using an string (key) and an int address (address of the field in namespaceInfo)?

Reply:
And then it could just be iterating over a big map of key -> int address